crates/telio-utils/src/tokio.rs

```diff
@@ -33,8 +34,15 @@ impl Default for ThreadTracker {
 
 impl ThreadTracker {
     fn set_status(&mut self, status: ThreadStatus) {
-        self.last_change = Instant::now();
-        self.statuses.insert(current().id(), status);
+        let now = Instant::now();
+        let delta = now - self.last_change;
+        self.last_change = now;
+        if let Some(ThreadStatus::Unparked) = self.statuses.insert(current().id(), status) {
+            if status == ThreadStatus::Parked && delta > UNPARKED_THRESHOLD {
+                let tid = std::thread::current().id();
+                telio_log_debug!("Thread {tid:?} was unparked for too long: {delta:?}");
+            }
+        }
```

Comment:
:thinking: if I understand correctly, this will not provide us specific information into what's blocking but rather the fact that it happens.(checks intervals between [unparked; parked]).

What do you think about a possible solution:
- Log when `tid` was `parked`. When `unpark`ing check how long it was actually parked meaning doing job we suspect might be too long.
- Hopefully stacktrace is useful in that case and we can track it down. Without stacktrace I suppose this is the same as existing implementation :)

The way I see it, in the current implementation we lose the information about the owner of the long task

Reply:
I had a wrong view of the park/unpark. Per my previous understanding parking thread meant _actually doing the work_. The whole comment is moot :)
In your case it's exactly how it should be it seems.